pkg/postgres/query/builder_test.go

```diff
@@ -273,7 +279,127 @@ func TestQueryBuilder(t *testing.T) {
+			wantQuery: `WHERE date_trunc('day'::text, "started_col_name") > date_trunc('day'::text, ?::timestamp) OR date_trunc('day'::text, "started_col_name") < date_trunc('day'::text, ?::timestamp) ORDER BY ? DESC OFFSET 2 LIMIT 5`,
+			wantArgs:  []any{"another_date", "some_date", "started_col_name"},
+		},
+		{
+			name: "build query with filter paging, sorting, greater than && less than operators",
+			mockArg: query.ResultSelector{
+				Filter: &filter.Request{
+					Fields: []filter.RequestField{
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsGreaterThan,
+							Value:    5.3,
+						},
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsLessThan,
+							Value:    8.2,
+						},
+					},
+					Operator: filter.LogicOperatorOr,
+				},
+				Paging: &paging.Request{
+					PageIndex: 2,
+					PageSize:  5,
+				},
+				Sorting: &sorting.Request{
+					SortColumn:    "started",
+					SortDirection: "desc",
+				},
+			},
+			wantQuery: `WHERE "severity_col_name" > ? OR "severity_col_name" < ? ORDER BY ? DESC OFFSET 2 LIMIT 5`,
+			wantArgs:  []any{5.3, 8.2, "started_col_name"},
+		},
+		{
+			name: "build query with filter paging, sorting, greater than && less than operators",
+			mockArg: query.ResultSelector{
+				Filter: &filter.Request{
+					Fields: []filter.RequestField{
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsGreaterThanOrEqualTo,
+							Value:    5.3,
+						},
+						{
+							Name:     "severity",
+							Operator: filter.CompareOperatorIsLessThanOrEqualTo,
+							Value:    8.2,
+						},
+					},
+					Operator: filter.LogicOperatorOr,
+				},
+				Paging: &paging.Request{
+					PageIndex: 2,
+					PageSize:  5,
+				},
+				Sorting: &sorting.Request{
+					SortColumn:    "started",
+					SortDirection: "desc",
+				},
+			},
+			wantQuery: `WHERE "severity_col_name" >= ? OR "severity_col_name" <= ? ORDER BY ? DESC OFFSET 2 LIMIT 5`,
+			wantArgs:  []any{5.3, 8.2, "started_col_name"},
+		},
+		{
+			name: "build query with filter paging, sorting, begins with && contains compare operators",
+			mockArg: query.ResultSelector{
+				Filter: &filter.Request{
+					Fields: []filter.RequestField{
+						{
+							Name:     "other_filter_field",
+							Operator: filter.CompareOperatorBeginsWith,
+							Value:    []any{"some text"},
+						},
+						{
+							Name:     "other_filter_field",
+							Operator: filter.CompareOperatorContains,
+							Value:    []any{"another text"},
+						},
+					},
+					Operator: filter.LogicOperatorOr,
+				},
+				Paging: &paging.Request{
+					PageIndex: 2,
+					PageSize:  5,
+				},
+				Sorting: &sorting.Request{
+					SortColumn:    "started",
+					SortDirection: "desc",
+				},
+			},
+			wantQuery: `WHERE ("other_filter_field_col_name" ILIKE ?) OR ("other_filter_field_col_name" ILIKE ?) ORDER BY ? DESC OFFSET 2 LIMIT 5`,
+			wantArgs:  []any{"some text", "another text", "started_col_name"},
```

Comment:
there is no difference in expectation between `filter.CompareOperatorBeginsWith` and `filter.CompareOperatorContains`, but there should be

Reply:
The `wantQuery` strings are now different.